### PR TITLE
Update cake build script for new tooling

### PR DESCRIPTION
**Pull request by slang25**

### CODE REVIEW

File: build.cake [hunk 1]
```diff
@@ -15,67 +18,90 @@ Task("Clean")
...
     .IsDependentOn("Build")
     .Does(() => {
-        DotNetCoreTest("./src/Shouldly.Tests/Shouldly.Tests.csproj");
+        DotNetCoreTool("./src/Shouldly.Tests/Shouldly.Tests.csproj", "xunit", "-configuration Debug");
+        try {
+            if (isAppVeyor)
+                AppVeyor.UploadTestResults("./src/Shouldly.Tests/report.xml", AppVeyorTestResultsType.XUnit);
+        } catch(Exception)
+        {
+
```
Comment (JosephWoodward) [inline build.cake]: Do we want to swallow this?

**Comment by JosephWoodward** [comment]
Looks like it's passing now, good job 👍 Just the one comment.

**Comment by slang25** [comment]
Sorry @JosephWoodward, only just noticed you had also started this, this was mostly from last night.

**Comment by JosephWoodward** [comment]
No worries, hadn't got very far. Wanted to have a play around with Cake anyway so it's all good.

File: build.cake (hunk 1 above)
Comment (slang25) [inline build.cake]: Yeah this is odd, was just doing this to debug. So it's actually throwing here, there is a `EnsureSuccessStatusCode` in the cake code, however the upload seems to work in that the tests all show up.

I'd like to know what is happening here, I'll do some more investigating.

File: build.cake [hunk 2]
```diff
@@ -15,67 +18,90 @@ Task("Clean")
...
-        // Update project.json
-        var updatedProjectJson = System.IO.File.ReadAllText(shouldlyProj)
-            .Replace("1.0.0-*", versionInfo.NuGetVersion);
-
-        System.IO.File.WriteAllText(shouldlyProj, updatedProjectJson);
+        
+        msBuildSettings = new DotNetCoreMSBuildSettings()
+                            .WithProperty("Version", versionInfo.NuGetVersion)
+                            .WithProperty("AssemblyVersion", versionInfo.AssemblySemVer)
+                            .WithProperty("FileVersion", versionInfo.NuGetVersion);
```
Comment (slang25) [inline build.cake]: What should FileVersion actually be?
Look in the build output at `Adding Environment Variable. name='GitVersion_` for a nice list of options

File: src/Shouldly/Shouldly.csproj [hunk 3]
```diff
@@ -3,22 +3,18 @@
...
-    <IncludeSymbols>True</IncludeSymbols>
-    <IncludeSource>True</IncludeSource>
-
-    <ComVisible>false</ComVisible>
-
+    <IncludeSymbols>true</IncludeSymbols>
+    <IncludeSource>true</IncludeSource>
     <AssemblyOriginatorKeyFile>sn.snk</AssemblyOriginatorKeyFile>
     <SignAssembly>true</SignAssembly>
+    <PublicSign Condition=" '$(OS)' != 'Windows_NT' ">true</PublicSign>
```
Comment (slang25) [inline src/Shouldly/Shouldly.csproj]: All the cool kids have this in their repo, I have no idea what it does... shhh

File: src/Shouldly/Shouldly.csproj [hunk 4]
```diff
@@ -3,22 +3,18 @@
...
+    <TargetFrameworks Condition=" '$(OS)' != 'Windows_NT' ">netstandard1.3</TargetFrameworks>
     <AssemblyName>Shouldly</AssemblyName>
     <PackageId>Shouldly</PackageId>
     <PackageTags>test;unit;testing;TDD;AAA;should;testunit;rspec;assert;assertion;framework</PackageTags>
     <PackageIconUrl>https://raw.github.com/shouldly/shouldly/master/Icons/package_icon.png</PackageIconUrl>
     <PackageProjectUrl>http://shouldly.github.com</PackageProjectUrl>
     <PackageLicenseUrl>https://github.com/shouldly/shouldly/blob/master/LICENSE.txt</PackageLicenseUrl>
-    <NetStandardImplicitPackageVersion>1.6.1</NetStandardImplicitPackageVersion>
-    
-    <GeneratePackageOnBuild>True</GeneratePackageOnBuild>
```
Comment (slang25) [inline src/Shouldly/Shouldly.csproj]: Took out `GeneratePackageOnBuild`, on macOS you can do a typical `dotnet build`, and if you want a package then run `.\build.sh`

File: src/Shouldly/Shouldly.csproj [hunk 5]
```diff
@@ -3,22 +3,18 @@
...
     <Description>Shouldly - Assertion framework for .NET. The way asserting *Should* be</Description>
     <TargetFrameworks>net451;net40;netstandard1.3</TargetFrameworks>
+    <TargetFrameworks Condition=" '$(OS)' != 'Windows_NT' ">netstandard1.3</TargetFrameworks>
     <AssemblyName>Shouldly</AssemblyName>
     <PackageId>Shouldly</PackageId>
     <PackageTags>test;unit;testing;TDD;AAA;should;testunit;rspec;assert;assertion;framework</PackageTags>
     <PackageIconUrl>https://raw.github.com/shouldly/shouldly/master/Icons/package_icon.png</PackageIconUrl>
     <PackageProjectUrl>http://shouldly.github.com</PackageProjectUrl>
     <PackageLicenseUrl>https://github.com/shouldly/shouldly/blob/master/LICENSE.txt</PackageLicenseUrl>
-    <NetStandardImplicitPackageVersion>1.6.1</NetStandardImplicitPackageVersion>
```
Comment (slang25) [inline src/Shouldly/Shouldly.csproj]: Took out this, cause apparently that's best practice

**Comment by slang25** [comment]
@JosephWoodward I've addressed the comment, we didn't need to upload the results file as AppVeyor does this for us somehow by magic.

Just need to make a decision on the `FileVersion` and we are good to go.

File: src/Shouldly/Shouldly.csproj (hunk 3 above)
Comment (JosephWoodward) [inline src/Shouldly/Shouldly.csproj]: lol, I did wonder what it was.

File: src/Shouldly/Shouldly.csproj (hunk 5 above)
Comment (JosephWoodward) [inline src/Shouldly/Shouldly.csproj]: Where did you read that?

**Comment by slang25** [comment]
@JosephWoodward this one is still ready to go, if you wanted a version out tonight this would be a good chance to test the deployment via AppVeyor.

File: src/Shouldly/Shouldly.csproj (hunk 5 above)
Comment (slang25) [inline src/Shouldly/Shouldly.csproj]: Haven't found the source of that remark.
I did find this:
https://github.com/shouldly/shouldly/pull/432#discussion_r125122510
That was mostly a objection to use `1.6.0` for the most part, I believe we get whatever the default for the sdk is, for 1.0.4 sdk = 1.6.1 implicit netstandard package. I think we should let it be implicit unless we understand a reason to specify it in order to not trip ourselves up in future.

**Pull request merged**